src/__tests__/environmentTree.ts

```diff
@@ -0,0 +1,66 @@
+import { createGlobalEnvironment, EnvTree, EnvTreeNode } from '../createContext'
+import { pushEnvironment } from '../interpreter/interpreter'
+import { mockContext, mockEnvironment } from '../mocks/context'
+
+test('EnvTree root should be null upon instantiation', () => {
+  const envTree = new EnvTree()
+  expect(envTree.root).toBeNull()
+})
+
+test('EnvTree::insert should insert the globalEnvironment as the root', () => {
+  const envTree = new EnvTree()
+  envTree.insert(createGlobalEnvironment())
+  expect(envTree.root).toMatchSnapshot()
```

Comment:
I think we should check whether it is === the env directly rather than using snapshots